test/metabase/test/redefs.clj

```diff
@@ -2,17 +2,50 @@
   "Redefinitions of vars from 3rd-party namespaces to make sure they do extra stuff we want (like initialize things if
   needed when running)."
   (:require
+   [clojure.pprint :as pprint]
+   [clojure.test :as t]
    [mb.hawk.parallel]
    [metabase.plugins.classloader :as classloader]
    [metabase.test.util.thread-local :as tu.thread-local]
+   [metabase.util.log :as log]
    [methodical.core :as methodical]
    [toucan2.connection :as t2.connection]
-   [toucan2.tools.with-temp :as t2.with-temp]))
+   [toucan2.core :as t2]
+   [toucan2.tools.with-temp :as t2.with-temp]
+   [toucan2.util :as t2.u]))
 
 (def ^:dynamic ^:private *in-tx*
   "Used to detect whether we're in a nested [[with-temp]]. Default is false."
   false)
 
+(def ^:dynamic ^:private *with-temp-clean-up*
+  "Whether with-temp should clean up afterward, default is true."
+  true)
+
+(methodical/defmethod t2.with-temp/do-with-temp* :default
```

Comment:
faithfully copied from `toucan2`, the only difference is in the finally clause

Reply:
This duplication is unfortunate. It looks to me like we could avoid it if we merely decorated `f`:

```clojure
(fn [temp-object]
  (try 
    (f temp-object)
    (finally 
      (when *with-temp-clean-up*
        ...))))
```

This just assumes that nothing in the `t/testing` call will throw, which seems fair. Having the clean-up inside this testing scope also seems fine.

Reply:
Oh nevermind, I should have looked at the base implementation first 🙈

Reply:
This duplication feels off to me, I feel like it's the lesser of two evils to add the dynamic var upstream.

This debugging flow seems like something generally useful to toucan2 users in anycase.